torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -9,12 +9,13 @@
 from torch.fx.experimental.refinement_types import Equality
 import itertools
 
-
 from torch.fx.experimental.unification import Var  # type: ignore[attr-defined]
 
+import sympy  # type: ignore[import]
```

Comment:
How about we do this: we do `try/except` and set `HAS_SYMPY` to True or False based on whether we can import it. Then we can pull the tuple at line 272 out as a constant like `DIMENSION_TYPES =  (int, sympy.Symbol) if HAS_SYMPY else (int,)`. Then we can wrap the definitions of `convert_to_sympy_symbols` and `infer_symbolic_relations` with `if HAS_SYMPY:`

In that case, I think if the user does not have sympy installed, this will degenerate into a still-workable API that just doesn't have the symbolic relations functions in it